build.gradle.kts

```
@@ -8,29 +8,30 @@ repositories {
 }
 
 dependencies {
-    implementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
-    implementation("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+    testImplementation(platform("org.junit:junit-bom:5.8.2"))
+    testImplementation("org.junit.jupiter:junit-jupiter")
 }
 
-tasks.test {
-    useJUnitPlatform()
+tasks.withType<Test> {
+    useJUnitPlatform() // JUnit 5
 }
 
 version = "0.1.1"
 group = "com.github.zeckie"
 
-// Work out current java version
+/*
+* Work out current java version
+* Note: versions prior to JEP223 (Java 9) had version numbers starting with "1." so
+* will be counted as major version 1.
+*/
 val MIN_JAVA_VER = 9
 val fullVersion = System.getProperty("java.version")
 val majorVersion = Integer.parseInt(fullVersion.substring(0, fullVersion.indexOf('.')))
 System.out.println("Java major version: " + majorVersion)
```

Using `Integer` in Kotlin code isn't idiomatic, and neither is using `System.out.println`.
```suggestion
val majorVersion = fullVersion.substringBefore('.').toInt()
println("Java major version: $majorVersion")
```

---

Anyway, it would be better to use Gradles builtin Java version api.

---

I'm sure there are lots of ways of doing this, and I'd prefer to do it more the Java way than the Kotlin way.

---

It's a Kotlin file...

Anyway, using `JavaVersion.current()` is a much better option.